ompi/mca/osc/ucx/osc_ucx_comm.c

```diff
@@ -280,6 +280,9 @@ static inline int get_dynamic_win_info(uint64_t remote_addr,
     }
 
     temp_buf = calloc(remote_state_len, 1);
+    if (NULL == temp_buf) {
+        return OMPI_ERR_OUT_OF_RESOURCE;
```

Comment:
`goto cleanup` is better, otherwise missing unlock in this case

Reply:
good catch, will update the PR, thx!

Reply:
Having a deeper look into the relevant code paths at the cleanup target the error propagation seems broken to me.

Only the error code returned by `ompi_osc_ucx_dynamic_unlock` is returned - and not previously reported errors that went there using goto. Guess this is not intended. 
Also, in case the acquiring of the lock fails, the unlock is performed - is this correct/necessary?

Diving deeper into the calltree there seems also a mix of OMPI and OPAL error codes: `ompi_osc_ucx_dynamic_unlock` initializes `ret = OMPI_SUCCESS` but is assigned later with the return value of `opal_common_ucx_wpmem_fetch` (returning OPAL_ERROR/OPAL_SUCCESS).

I think this will take me a moment to sort out and prepare an updated PR.

Reply:
please check #11028, it is already covered there

Reply:
Thanks for pointing out. But #11028 seems not to cover all the issues above for me.
I updated this PR addressing all mentioned points from above. I separated it into logical commits for all the different pieces.